applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationAsyncClientBuilder.java

```diff
@@ -98,10 +99,12 @@ public ConfigurationAsyncClient build() {
         policies.add(new AddDatePolicy());
         policies.add(new ConfigurationCredentialsPolicy());
         policies.add(new AsyncCredentialsPolicy(credentials));
+        HttpPolicyProviders.addBeforeRetryPolicies(policies);
```

Comment:
How did we determine that the policies be added before and after the retryPolicy? For the Configuration service, the Authorization header value is determined by the contents of the request body and some of the headers. If a policy modifies the request body or dependent header values that it's possible that the signature is no longer valid.

Reply:
The general discussion we've had is that these are the two places where policies need to be added, as some things need to happen on every retry, and some do not. Ordering of policies is poorly defined and I don't have a good answer here. I feel like we should add more API here to allow for priorities to be specified. I chose to put the 'before retry policies' right at the end before the retry, but that was an arbitrary choice.

Reply:
>The general discussion we've had is that these are the two places where policies need to be added, as some things need to happen on every retry, and some do not.

does it make sense to define an interface for RetryPolicy like:

```java
public interface RetryPolicy extends HttpPipelinePolicy {
}
```

All retry policy implementations implements this interface.

HttpPolicyProviders.addBeforeRetryPolicies and HttpPolicyProviders.addAfterRetryPolicies can iterate through input `List<HttpPipelinePolicy>` and locate the policy for retry (that implements RetryPolicy) then add dynamic policies before or after it.